.github/workflows/ci.yml

```
@@ -224,6 +251,20 @@ jobs:
     name: Build Image
     runs-on: ubuntu-22.04
     needs: [vars, binary]
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - dockerfile: build/Dockerfile
+            image: ghcr.io/nginxinc/nginx-kubernetes-gateway
+            target: goreleaser
+            sarif-file: trivy-results-nginx-kubernetes-gateway.sarif
+          - dockerfile: build/Dockerfile.nginx
+            image: ghcr.io/nginxinc/nginx-kubernetes-gateway/nginx
+            sarif-file: trivy-results-nginx-kubernetes-gateway-nginx.sarif
+            build-args: |
+              NJS_DIR=internal/mode/static/nginx/modules/src
+              NGINX_CONF_DIR=internal/mode/static/nginx/conf
```

I would simplify the matrix with something like
```suggestion
      matrix:
        container: [nkg, nginx]
```
and then use `matrix.container` like for the container to conditionally add `nginx`
```
name=ghcr.io/nginxinc/nginx-kubernetes-gateway${{ matrix.container == 'nginx' && '/nginx' || '' }}}
```

it would also make it easier to add a scope in the cache (if you have the same scope it will overwrite and actually never use it) `cache-from: type=gha,scope=${{ matrix.container}}` without adding an additional field to the matrix.

Also you can leave the `build-args`, it doesn't have any effect for the other container if they're not in the Dockerfile.